hera_sim/visibilities/simulators.py

```diff
@@ -13,374 +12,129 @@
-    sky_freqs : array_like, optional
-        Frequencies at which the sky intensity and/or point sources
-        are defined in [Hz]. Defaults to the unique frequencies in
-        `uvdata` Shape=(NFREQS,).
-    beams : array_like of `pyuvsim.analyticbeam.AnalyticBeam`, optional
+    sky_model
+        A model for the sky to simulate.
+    beams
         UVBeam models for as many antennae as have unique beams.
         Initialized from `obsparams`, if included. Defaults to a
         single uniform beam is applied for every antenna. Each beam
         is the response of an individual antenna and NOT a
         per-baseline response.
         Shape=(N_BEAMS,).
-    beam_ids : array_like of int, optional
+    beam_ids
         List of integers specifying which beam model each antenna
         uses (i.e. the index of `beams` which it should refer to).
         Initialized from `obsparams`, if included. By default, all
         antennas use the same beam (beam 0).
         Shape=(N_ANTS,).
-    sky_intensity : array_like, optional
-        A healpix model for the intensity of the sky emission, in
-        [Jy/sr]. Shape=(NFREQS, N_PIX_SKY).
-    point_source_pos : array_like, optional
-        An array of point sources. For each source, the entries are
-        (ra, dec) [rad] (assumed to be in J2000).
-        Shape=(N_SOURCES, 2).
-    point_source_flux : array_like, optional
-        An array of fluxes of the given point sources, per
-        frequency. Fluxes in [Jy]. Shape=(NFREQS, N_SOURCES).
-    nside : int, optional
-        Only used if sky_intensity is *not* given but the simulator
-        is incapable of directly dealing with point sources. In this
-        case, it sets the resolution of the healpix map to which the
-        sources will be allocated.
 
     Notes
     -----
-        Input beam models represent the responses of individual
-        antennas and are NOT the same as per-baseline "primary
-        beams". This interpretation of a "primary beam" would be the
-        product of the responses of two input antenna beams.
+    Input beam models represent the responses of individual
+    antennas and are NOT the same as per-baseline "primary
+    beams". This interpretation of a "primary beam" would be the
+    product of the responses of two input antenna beams.
     """
 
-    #: Whether this particular simulator has the ability to simulate point
-    #: sources directly.
-    point_source_ability = True
-
-    #: Whether this particular simulator has the ability to simulate diffuse
-    #: maps directly.
-    diffuse_ability = True
-
     def __init__(
         self,
-        obsparams=None,
-        uvdata=None,
-        sky_freqs=None,
-        beams=None,
-        beam_ids=None,
-        sky_intensity=None,
-        point_source_pos=None,
-        point_source_flux=None,
-        nside=2 ** 5,
-        validate=True,
+        *,
+        uvdata: UVData,
+        sky_model: SkyModel,
```

Comment:
The idea is to make sure they're always passed as keywords. This is good for future-proofing the API (in case we change parameter ordering in any way).

Reply:
Ah, gotcha! I think this is a very good move, then.